### PR TITLE
Get package as T4 Manifest

### PR DESCRIPTION
**Pull request by kevinemoore**
## Description
Adding a view to get a package from the registry as a T4 manifest.

## TODO

Use this section for work-in-progress pull requests. If you're using this section,
please tag your PR "WIP". 

- [ ] Unit Tests
- [x] Add node metadata to manifest

## Calvin TODO

- [ ] Add role info to `list_users`

### CODE REVIEW

File: registry/quilt_server/views.py [hunk 1]
```diff
@@ -2339,3 +2431,38 @@ def comments_list(owner, package_name):
...
     return dict(comments=map(_comment_dict, comments))
+
+IDENTITY_POOL_ID = 'us-east-1:bad5e611-6efb-4b71-8f46-674fa56901b8'
+AWS_TOKEN_DURATION = 60 * 60 # 60 minutes
+
+@app.route('/api/auth/get_credentials', methods=['GET'])
+@api(require_login=True)
+@as_json
+def aws_token():
+    client = boto3.client('cognito-identity')
```
Comment (dimaryaz) [inline registry/quilt_server/views.py]: Need to pass credentials here, the way we're doing it for the `s3_client`. Don't want to accidentally access the real thing while running unit tests.
(And it should probably be a global, too.)

File: registry/quilt_server/views.py [hunk 2]
```diff
@@ -1033,6 +1035,89 @@ def package_get(owner, package_name, package_hash):
...
+                Key='%s/%s/%s' % (OBJ_DIR, owner, node.metadata_hash)
+                )
+
+            body = resp['Body']
+            encoding = resp.get('ContentEncoding')
+            data = body.read()
+            node_meta = json.loads(data)
+        else:
+            node_meta = None
+        t4pkg.set(key, t4.packages.PackageEntry(pkeys, node_size, hash_objs, meta=node_meta))
```
[the remark this answers is not in the document]
Comment (kevinemoore) [reply registry/quilt_server/views.py]: That might be true, but importing T4 here feels safer. We don't want to have an undocumented dependency on the T4 registry format here.

File: registry/quilt_server/views.py [hunk 3]
```diff
@@ -1033,6 +1035,89 @@ def package_get(owner, package_name, package_hash):
...
+    urls = {
+        blob_hash: _generate_presigned_url(S3_GET_OBJECT, owner, blob_hash)
+        for blob_hash in all_hashes
+    }
+
+    # Translate Quilt 2 package tree to T4 manifest
+    t4pkg = t4.Package()
+    for node, key in _iterate_data_nodes_with_path(instance.contents, ''):
+        blobs = [blob_by_hash[blob_hash] for blob_hash in node.hashes]
+        pkeys = ['s3://%s/%s/%s/%s' % (PACKAGE_BUCKET_NAME,
```
[the remark this answers is not in the document]
Comment (kevinemoore) [reply registry/quilt_server/views.py]: The physical keys will be inaccessible so the client will need to have the registry sign the URLs. Or, we could return signed URLs as the physical keys, but the T4 client doesn't support URLs as physical keys yet.

**Comment by dimaryaz** [comment]
Does the manfest rely on roles? If not, can you split it up into multiple PRs?

Ideally:
- T4 Manfiest
- Credentials and roles
- Dockerfile cleanup

**Comment by dimaryaz** [comment]
Actually, @meffij : could you split it up?

**Comment by meffij** [comment]
yes I will create a new PR with just the role stuff

**Comment by meffij** [comment]
new role PR here https://github.com/quiltdata/quilt/pull/801

**Pull request closed without merging**